packages/connect/src/device/checkFirmwareRevision.ts

```
@@ -11,13 +11,21 @@ type GetOnlineReleaseMetadataParams = {
     internalModel: string;
 };
 
+const OFFLINE_ERROR = 'OFFLINE_ERROR';
+
 const getOnlineReleaseMetadata = async ({
     firmwareVersion,
     internalModel,
 }: GetOnlineReleaseMetadataParams): Promise<FirmwareRelease | undefined> => {
-    const onlineReleases = await downloadReleasesMetadata({ internal_model: internalModel });
-
-    return onlineReleases.find(onlineRelease => isEqual(onlineRelease.version, firmwareVersion));
+    try {
+        const onlineReleases = await downloadReleasesMetadata({ internal_model: internalModel });
+
+        return onlineReleases.find(onlineRelease =>
+            isEqual(onlineRelease.version, firmwareVersion),
+        );
+    } catch {
+        throw OFFLINE_ERROR;
```

This will hide original error.

---

Also I do not really like throwing something what is not the exception (in JS the `Error`)

---

The original implementation did not use the error either :shrug: 
It only reported `failFirmwareRevisionCheck('other-error')` or `failFirmwareRevisionCheck('cannot-perform-check-offline')`